From fcf4dc885f73183cefd04cd61a1246bf62b31274 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 04 Jan 2021 00:12:55 +0200
Subject: [PATCH] [SW-28128] gaudi: security: close lbw from dma

The DMA LBW access is not routed through the regular routers and
therefore does not abey their settings. Instead, it has its own set of
the RRs to be restricted by.
Close the LBW privilege areas access from DMA.

Change-Id: Ic5fb3a2ccda83e8133e7e791e420ea9bb6c2529a
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 8fa4dd5..49f1c01 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1065,6 +1065,8 @@
 enum preboot_sec_lbw_rr_type {
 	IF_RTR,
 	DMA_IF_SOB,
+	DMA_IF_CH0,
+	DMA_IF_CH1,
 };
 
 #define DMA_IF_NUM	4
@@ -1090,6 +1092,26 @@
 		.max_rd_breg = mmDMA_IF_W_S_SOB_MAX_RPRIV_0,
 		.hit_wr_breg = mmDMA_IF_W_S_SOB_HIT_WPRIV,
 		.hit_rd_breg = mmDMA_IF_W_S_SOB_HIT_RPRIV,
+	},
+	[DMA_IF_CH0] = {
+		.instances_num = DMA_IF_NUM,
+		.instances_dist = DMA_IF_DIST,
+		.min_wr_breg = mmDMA_IF_W_S_DMA0_MIN_WPRIV_0,
+		.max_wr_breg = mmDMA_IF_W_S_DMA0_MAX_WPRIV_0,
+		.min_rd_breg = mmDMA_IF_W_S_DMA0_MIN_RPRIV_0,
+		.max_rd_breg = mmDMA_IF_W_S_DMA0_MAX_RPRIV_0,
+		.hit_wr_breg = mmDMA_IF_W_S_DMA0_HIT_WPRIV,
+		.hit_rd_breg = mmDMA_IF_W_S_DMA0_HIT_RPRIV,
+	},
+	[DMA_IF_CH1] = {
+		.instances_num = DMA_IF_NUM,
+		.instances_dist = DMA_IF_DIST,
+		.min_wr_breg = mmDMA_IF_W_S_DMA1_MIN_WPRIV_0,
+		.max_wr_breg = mmDMA_IF_W_S_DMA1_MAX_WPRIV_0,
+		.min_rd_breg = mmDMA_IF_W_S_DMA1_MIN_RPRIV_0,
+		.max_rd_breg = mmDMA_IF_W_S_DMA1_MAX_RPRIV_0,
+		.hit_wr_breg = mmDMA_IF_W_S_DMA1_HIT_WPRIV,
+		.hit_rd_breg = mmDMA_IF_W_S_DMA1_HIT_RPRIV,
 	},
 };
 
@@ -1129,6 +1151,8 @@
 {
 	preboot_secure_lbw_rr_gen(rr, min, max, IF_RTR);
 	preboot_secure_lbw_rr_gen(rr, min, max, DMA_IF_SOB);
+	preboot_secure_lbw_rr_gen(rr, min, max, DMA_IF_CH0);
+	preboot_secure_lbw_rr_gen(rr, min, max, DMA_IF_CH1);
 	preboot_secure_pcie_rr(rr, min, max);
 }
 
